File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency2/travelAgency2/Repository/Database/TableFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.SQLite;
using System.Configuration;
using travelAgency2.Utils;
using travelAgency2.src.Domain;

namespace travelAgency2.Repository.Database
{
    internal class TableFactory
    {
        private SQLiteConnection connection;
        public TableFactory()
        {
            string config = ConfigurationManager.AppSettings.Get("db");
            this.connection = new SQLiteConnection(config);
            try
            {
                connection.Open();
                Console.WriteLine("open");
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
            }
        }

        public object getTable(Constants.Db.Tables table)
        {
            if (table == Constants.Db.Tables.USER)
                return new UserTable(connection);
            if (table == Constants.Db.Tables.FLIGHT)
                return new FlightTable(connection);
            if (table == Constants.Db.Tables.PURCHASE)
                return new PurchaseTable(connection);
            return new LocationTable(connection);
    }
}

}

File: second-year/second-semester/MPP/mpp-proiect-repository-raulbaciulescu/travelAgency/Persistence/src/main/java/impl/database/TableFactory.java
package impl.database;

import api.Constants;
import api.Table;
import org.springframework.stereotype.Component;

import java.io.FileReader;
import java.io.IOException;
import java.sql.SQLException;
import java.util.Properties;

@Component
public class TableFactory {
    private final JdbcUtils dbUtils;

    public TableFactory() {
        Properties properties = new Properties();
        try {
            properties.load(new FileReader(Constants.Db.filename));
        } catch (IOException e) {
            System.out.println("Cannot find bd.config " + e);
        }
        dbUtils = new JdbcUtils(properties);
    }

    public Table<?, ?> getTable(Constants.Db.Tables table) throws SQLException {
        if (table == Constants.Db.Tables.USER)
            return new UserTable(dbUtils.getConnection());
        if (table == Constants.Db.Tables.FLIGHT)
            return new FlightTable(dbUtils.getConnection());
        if (table == Constants.Db.Tables.PURCHASE)
            return new PurchaseTable(dbUtils.getConnection());
        if (table == Constants.Db.Tables.FLIGHT_NEW)
            return new FlightNewTable(dbUtils.getConnection());

        return new LocationTable(dbUtils.getConnection());
    }
}
